jdaviz/configs/default/plugins/metadata_viewer/metadata_viewer.py

```diff
@@ -47,14 +48,15 @@ def __init__(self, *args, **kwargs):
 
     @property
     def user_api(self):
-        return PluginUserApi(self, expose=('dataset', 'show_primary'), readonly=('metadata',))
+        return PluginUserApi(self, expose=('dataset', 'show_primary'), readonly=('meta',))

```

Comment:
`metadata` was public API, so maybe we should keep and deprecate?  Is the change in name only because the change in format?

Reply:
> deprecate

Perhaps. Depends on whether people insist of changing `metadata` type itself or not during review. I will hold off until we have agreement. Thanks.

Reply:
> metadata as a list is used in Vue for actual displaying as well. Changing metadata itself from list to dict would require more refactoring on the front end.

The only potential difficult part might be sorting, but OrderedDict might do the trick if traitlets accept them.

But we can also always have the internal attribute renamed to something just for internal use (`metadata_list` or something).

Reply:
There is a `Dict` traitlet but I don't know exactly how it is going to behave. Because we also grab the header comments, we still need the list for front end anyway, so `metadata_list` might be the way to go. Still need more refactoring though. I can look into that if you prefer that over deprecating `metadata` from public API. Lemme know.